### PR TITLE
Host require minimum number of admins for Collectives

### PR DESCRIPTION
**Pull request by kewitz**
Requires https://github.com/opencollective/opencollective-api/pull/7521
Resolves https://github.com/opencollective/opencollective/issues/3593

### CODE REVIEW

**Comment by kewitz** [comment]
Rebased.

File: components/ApplyToHostModal.js [hunk 1]
```diff
@@ -53,14 +56,49 @@ const hostFields = gqlV2/* GraphQL */ `
...
+          role
+          account {
+            id
+            type
+            slug
+            name
+            imageUrl
+          }
+        }
+      }
```
Comment (znarf) [inline components/ApplyToHostModal.js:100]: Use fragment?

File: components/edit-collective/sections/Policies.js [hunk 2]
```diff
@@ -153,23 +164,25 @@ const Policies = ({ collective, showOnlyExpensePolicy }) => {
...
-        variables: {
-          account: {
-            legacyId: collective.id,
+        }),
+        setPolicies({
+          variables: {
+            account: {
+              legacyId: collective.id,
+            },
+            policies: omitDeep(policies, ['__typename']),
```
Comment (znarf) [inline components/edit-collective/sections/Policies.js:182]: move `omitDeep` on top?

File: components/edit-collective/sections/Policies.js [hunk 3]
```diff
@@ -204,10 +217,19 @@ const Policies = ({ collective, showOnlyExpensePolicy }) => {
...
     }
   }, [data]);
 
+  const numberOfAdminsOptions = [0, 2, 3, 4, 5].map(n => ({
+    value: n,
+    label: formatMessage(messages['requiredAdmins.numberOfAdmins'], { admins: n }),
+  }));
+  const minAdminsApplies = [
+    { value: 'NEW_COLLECTIVES', label: <FormattedMessage defaultMessage="New Collectives Only" /> },
+    { value: 'ALL_COLLECTIVES', label: <FormattedMessage defaultMessage="All Collectives" /> },
```
Comment (znarf) [inline components/edit-collective/sections/Policies.js:230]: disable ALL_COLLECTIVES for now?

File: components/ocf-host-application/ApplicationForm.js [hunk 4]
```diff
@@ -539,6 +555,82 @@ const ApplicationForm = ({
                           />
                         </P>
                       </Box>
+                      <Box width={['256px', '484px', '663px']} my={2}>
+                        <P fontSize="13px" lineHeight="16px" color="#4E5052">
+                          Add Administrators
```
Comment (znarf) [inline components/ocf-host-application/ApplicationForm.js:560]: Localization ?

File: components/onboarding-modal/OnboardingModal.js [hunk 5]
```diff
@@ -428,6 +431,36 @@ const addEditCollectiveContactMutation = graphql(editCollectiveContactMutation,
...
+            name
+            imageUrl
+            slug
+          }
+        }
+      }
+    }
+  `,
+  {
+    name: 'memberInvitationsQuery',
```
Comment (znarf) [inline components/onboarding-modal/OnboardingModal.js:452]: Is that useful? Remove if not.

File: components/onboarding-modal/OnboardingModal.js [hunk 6]
```diff
@@ -359,6 +361,7 @@ class OnboardingModal extends React.Component {
                             values={values}
                             errors={errors}
                             touched={touched}
+                            memberInvitations={memberInvitationsQuery?.account?.memberInvitations || []}
```
Comment (znarf) [inline components/onboarding-modal/OnboardingModal.js:364]: data.memberInvitations would look slightly simpler?

File: components/onboarding-modal/OnboardingModal.js (hunk 5 above)
Comment (znarf) [reply components/onboarding-modal/OnboardingModal.js:452]: This is not a pattern we're using usually.

**Review by znarf — COMMENTED** [review]
Because there is a lot of new strings, check that we're not introducing duplicates.

File: pages/ocf-host-application.js [hunk 7]
```diff
@@ -35,6 +35,20 @@ const collectiveQuery = gqlV2/* GraphQL */ `
   }
 `;
 
+const hostQuery = gqlV2/* GraphQL */ `
+  query HostData {
```
Comment (znarf) [inline pages/ocf-host-application.js:39]: should be consistent with the variable name and try to be unique, explaining the context

File: pages/ocf-host-application.js [hunk 8]
```diff
@@ -35,6 +35,20 @@ const collectiveQuery = gqlV2/* GraphQL */ `
   }
 `;
 
+const hostQuery = gqlV2/* GraphQL */ `
```
Comment (znarf) [inline pages/ocf-host-application.js:38]: `ocfHostApplicationPageQuery`

File: pages/ocf-host-application.js (hunk 7 above)
Comment (znarf) [reply pages/ocf-host-application.js:39]: or combine with the other query.

File: components/create-collective/CreateCollectiveForm.js [hunk 9]
```diff
@@ -281,6 +286,79 @@ class CreateCollectiveForm extends React.Component {
                       <P fontSize="11px" color="black.600">
                         {intl.formatMessage(messages.descriptionHint)}
                       </P>
+                      {host && (
+                        <Box mt={3} mb={2}>
+                          <P {...LABEL_STYLES}>Add Administrators</P>
```
Comment (znarf) [inline components/create-collective/CreateCollectiveForm.js:291]: Localization

**Pull request merged**